File: NeuralNetNative/ConstantFactor.cpp
#define NEURALNETNATIVEAPI
#include "ConstantFactor.h"

namespace NeuralNetNative {
	ConstantFactor::ConstantFactor(float constantValue) {
		_constantValue = constantValue;
	}
	
	float ConstantFactor::GetFactor(int iterNumber) const {
		return _constantValue;
	}
}

File: StandardTypes/FactorStrategy/ConstantFactor.cs
namespace StandardTypes.FactorStrategy {
	public sealed class ConstantFactor : IFactorStrategy {
		private readonly float _constValue;
		
		public ConstantFactor(float constValue = 1f) {
			_constValue = constValue;
		}

		public float ConstantValue {
			get { return _constValue; }
		}

		public float GetFactor(int iterNumber) {
			return _constValue;
		}

		public float GetFactor() {
			return _constValue;
		}
	}
}
